logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -83,7 +92,10 @@
 import static org.logstash.instrument.metrics.UptimeMetric.ScaleUnits.SECONDS;
 
 /**
- * JRuby extension to provide ancestor class for Ruby's Pipeline and JavaPipeline classes.
+ * JRuby extension to provide ancestor class for the ruby-defined {@code LogStash::JavaPipeline} class.
+ *
+ * <p>NOTE: Although this class' name implies that it is "abstract", we instantiated it directly
+ *          as a lightweight temporary-scoped pipeline in the ruby-defined {@code LogStash::PipelineAction::Reload}
  * */
 @JRubyClass(name = "AbstractPipeline")
 public class AbstractPipelineExt extends RubyBasicObject {
```

Comment:
I am leaning toward `BasePipeline`, since it is not abstract. After we have a green CI pass here, I will perform another automated-rename in a separate commit.

Reply:
I do not have any specific plans for the cleanup of this class and rearrangement. This PR is one step to remove the hierarchy, and everything else is out of scope.